save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt

```diff
@@ -164,8 +164,7 @@ class VulnerabilityService(
                     } ?: cb.and()
 
                     val tagJoin: Join<LnkVulnerabilityTag, Tag> = root.join("tag")
-                    val inExpression = cb.`in`(tagJoin.get<String>("name"))
-                    val tagPredicate = inExpression.`in`(tags)
+                    val tagPredicate = cb.`in`(tagJoin.get<String>("name")).apply { tags.forEach(::value) }
```

Comment:
why do you need this duplicate code?
Maybe you can do it simply:
```val tagPredicate = if (filters.tags.isEmpty()) else cb.`in`(tagJoin.get<String>("name")).apply { tags.forEach(::value) }```

Reply:
The point is that if there are tags in filters, we need to interact with `lnk_vulnerability_tag` table, while `vulnerability` should be used if no tags are set

Reply:
And, these are different repositories, I understand.
I think then you can still do something like this:
val newRoot = if (filters.tags.isEmpty()) root else root.join("tags")

Reply:
Fixed - take a look at my gpt4 solution and comment pls

Reply:
Thank you! Looks great :)